astropy/table/table.py

```diff
@@ -469,10 +475,13 @@ def filled(self, fill_value=None):
             New table with masked values filled
         """
         if self.masked:
+            # Get new columns with masked values filled, then create Table with those
+            # new cols (copy=False) but deepcopy the meta.
             data = [col.filled(fill_value) for col in self.columns.values()]
```

Comment:
This was previously doing the filling (which makes a new copy of filled data) and then copying that again (since the default Table init copies the data).  So short-circuit here and don't copy again.